main.nf

```diff
@@ -9,279 +9,812 @@
+
+    output:
+    file("rates/*csv") into alleyoopRatesOut
+
+    when:
+    !params.quantseq
+
+    script:
+    """
+    alleyoop rates -o rates \
+       -r ${fasta} \
+       -mq ${params.base_quality} \
+       -t ${task.cpus} \
+       ${filter[0]}
+    """
+}
+
+/*
+* STEP 8 - utrrates
+*/
+process utrrates {
+
+    label 'slamdunk_process'
+
+    tag { name }
+
+    input:
+    set val(name), file(filter), file(snp) from slamdunkForUtrRatesChannel
+    each file(fasta) from fastaUtrRatesChannel
+    each file(bed) from utrratesChannel
+
+    output:
+    file("utrrates/*csv") into alleyoopUtrRatesOut
+
+    when:
+    !params.quantseq
+
+    script:
+    """
+    alleyoop utrrates -o utrrates \
+       -r ${fasta} \
+       -mq ${params.base_quality} \
+       -b ${bed} \
+       -l ${params.read_length} \
+       -t ${task.cpus} \
+       ${filter[0]}
+    """
+}
+
+/*
+* STEP 9 - tcperreadpos
+*/
+process tcperreadpos {
+
+    label 'slamdunk_process'
+
+    tag { name }
+
+    input:
+    set val(name), file(filter), file(snp) from slamdunkForTcPerReadPosChannel
+    each file(fasta) from fastaReadPosChannel
+
+    output:
+    file("tcperreadpos/*csv") into alleyoopTcPerReadPosOut
+
+    when:
+    !params.quantseq
+
+    script:
+    snpMode = params.vcf ? "-v ${params.vcf}" : "-s . "
+    """
+    alleyoop tcperreadpos -o tcperreadpos \
+       -r ${fasta} \
+       ${snpMode} \
+       -mq ${params.base_quality} \
+       -l ${params.read_length} \
+       -t ${task.cpus} \
+       ${filter[0]}
+    """
+}
+
+/*
+* STEP 10 - tcperutrpos
+*/
+process tcperutrpos {
+
+    label 'slamdunk_process'
+
+    tag { name }
+
+    input:
+    set val(name), file(filter), file(snp) from slamdunkForTcPerUtrPosChannel
+    each file(fasta) from fastaUtrPosChannel
+    each file(bed) from utrposChannel
+
+    output:
+    file("tcperutrpos/*csv") into alleyoopTcPerUtrPosOut
+
+    when:
+    !params.quantseq
+
+    script:
+    snpMode = params.vcf ? "-v ${params.vcf}" : "-s . "
+    """
+    alleyoop tcperutrpos -o tcperutrpos \
+       -r ${fasta} \
+       -b ${bed} \
+       ${snpMode} \
+       -mq ${params.base_quality} \
+       -l ${params.read_length} \
+       -t ${task.cpus} \
+       ${filter[0]}
+    """
+}
+
+slamdunkFilterSummary
+   .flatten()
+   .filter( ~/.*bam$/ )
+   .collect()
+   .set { slamdunkFilterSummaryCollected }
+
+slamdunkCountAlleyoop
+   .collect()
+   .flatten()
+   .filter( ~/.*tsv$/ )
+   .collect()
+   .set{ slamdunkCountAlleyoopCollected }
+
+/*
+* STEP 11 - Summary
+*/
+process summary {
+
+    input:
+    file("filter/*") from slamdunkFilterSummaryCollected
+    file("count/*") from slamdunkCountAlleyoopCollected.ifEmpty([])
+
+    output:
+    file("summary*.txt") into summaryQC
+
+    script:
+    if (params.quantseq) {
+      """
+      alleyoop summary -o summary.txt ./filter/*bam
+      """
+    } else {
+      """
+      alleyoop summary -o summary.txt -t ./count ./filter/*bam
```

Comment:
Any reason not to put this in a variable with a single script block, like in other processes?

Reply:
Nope ;)